backend/src/process.py

```diff
@@ -440,13 +448,19 @@ def __create_node_finish(self, node_id: NodeId) -> Event:
         if not node_id in finished:
             finished.append(node_id)
 
+        self.completed_node_ids.add(node_id)
+
+        total_length = len(self.chain.nodes.keys())
```

Comment:
I don't think we need the `.keys()`. Same for the other 2 occurrences.

```suggestion
        total_length = len(self.chain.nodes)
```